.ci/Jenkinsfile

```diff
@@ -58,18 +59,57 @@ pipeline {
               from: "${env.GIT_PREVIOUS_SUCCESSFUL_COMMIT}",
               to: "${env.GIT_COMMIT}",
               patterns: regexps)
+
+            // regexp for JSON specs
+            regexps =[
+              "^tests/agents/json-specs/.*"
+            ]
+            env.JSON_SPECS_UPDATED = isGitRegionMatch(
+              from: "${env.GIT_PREVIOUS_SUCCESSFUL_COMMIT}",
+              to: "${env.GIT_COMMIT}",
+              patterns: regexps)
           }
         }
         stash allowEmpty: true, name: 'source', useDefaultExcludes: false
       }
     }
-    stage('Send Pull Request'){
+    stage('Send Pull Request for BDD specs'){
+      options {
+        warnError('Pull Requests to APM agents failed')
+      }
+      when {
+        beforeAgent true
+        anyOf {
+          expression { return env.GHERKIN_SPECS_UPDATED != "false" }
```

Comment:
I'm not sure I understand why this is `!= false` instead of just `true`.

Reply:
It was there before this changes, as a result of reusing this logic from another pipeline. I like the bool check you suggest the most, as it's clearer to read

Reply:
Updated in new commits, thanks!